File: 0058-length-of-last-word/0058-length-of-last-word.cpp
/*
    Approach: 
    Traverse from end to the first whitespace character and count the number of letters.
    Return the count as our pointer hits the whitespace character.
    
    Time complexity: O(n)
    Space complexity: O(1)
    */    
class Solution {
public:
    int lengthOfLastWord(string s) {
        
        int n=s.length();
        
        int i = n-1;
        int len = 0;
        
        while(s[i] == ' '){
            i--;
            
        }
        while(i>=0 && s[i] !=' '){
            len++;
            i--;
        }
        
        return len;
        
    }
};

/*
int main() {
    Solution solution;
    string s = "Hello World";

    int result = solution.lengthOfLastWord(s);

    cout << "Length of the last word: " << result << endl;

    return 0;
}
*/

File: 0058-length-of-last-word/0058-length-of-last-word.py

    ##Approach: 
    ##Traverse from end to the first whitespace character and count ##the number of letters.
    ##Return the count as our pointer hits the whitespace character.
    
    ##Time complexity: O(n)
    ##Space complexity: O(1)  
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i=len(s) - 1
        length = 0
        
        while(s[i] == " "):
            i-=1
        while(i>=0 and s[i] != " "):
            length+=1
            i-=1
        return length    
    
    
#solution  = Solution()
#s="hello world"

#result = solution.lenghtOfLastWord(s)
#print(result)